### DISCUSSION TITLE
Conditional formatting (ternaries) and syntax

### DISCUSSION THREAD

**Discussion by Chris-Bee** [retorquere/zotero-better-bibtex#2666] [Q&A] [answered]
Hello,

thanks for the great plugin!

I'm trying to format a citation key depending on the existence of the _shorttitle_ field. I did find the [documentation](https://retorque.re/zotero-better-bibtex/citing/) for ternaries, but the syntax described there does not seem to work, at least when I use it for Zotero 6.0.27 and Better BibTex 6.7.122

This format works for my version:

```
[auth:capitalize] [year] [ShortTitle:lower]
```
What I'd like todo is to introduce an underscore with the shorttitle only if the shorttitle does excist. With the described syntax, this would be:

```
(auth:capitalize) (year) (ShortTitle ? "_") (ShortTitle:lower)
```

But this seems to be the wrong syntax, at least the evaluation reports "faild to upgrade legacy formular". I also saw examples from others using syntax like:

```
auth.capitalize + year
```

But this actually prints this text as the citation key.

Did the syntax change or am I missing something?

Thanks in advance.

**Comment by retorquere** [comment, marked as answer]
There's a few things going on here:

`:` from the old format has been replaced by `.` in the new syntax, so `[ShortTitle:lower]` becomes `ShortTitle.lower` (or `(ShortTitle.lower)` if you want the parenthesis). If you simply enter your formula in the old syntax, BBT will convert it into the new syntax for you (`auth(n=0,m=1,creator="*",initials=false).fold.capitalize + year + ShortTitle.lower`, which simplifies to `auth.capitalize + year + ShortTitle.lower`)

the ternary syntax is `(<condition> ? <what to include if the condition is non-empty> : <what to include if the condition is empty>)`; `(condition ? X) (condition : X)` will not work, you need `(ShortTitle ? "_" : "") + ShortTitle.lower`

What you want is easier achieved as `auth.capitalize + year + ShortTitle.prefix('_')` though.

**Reply by Chris-Bee** [reply]
Works like a charm. Thank you!
